nuxeo-services/nuxeo-platform-types-core/src/main/java/org/nuxeo/ecm/platform/types/TypeRegistry.java

```diff
@@ -242,7 +243,41 @@ public boolean hasType(String id) {
     }
 
     public Type getType(String id) {
-        return types.get(id);
+        Type type = types.get(id);
+        if (type == null) {
+            return null;
+        } else {
+            Map<String, SubType> allowedSubTypes = getCoreAllowedSubtypes(type);
+            Type newType = type.clone();
+            newType.setAllowedSubTypes(allowedSubTypes);
+            newType.setDeniedSubTypes(new String[0]);
+            return newType;
```

Comment:
maybe add a comment in the code here, explaining that denied sub types are already filtered from the allowed sub types by the core API (? if that's indeed the explanation :))